server/eclipse-project/src/main/java/us/freeandfair/corla/hibernate/Persistence.java

```diff
@@ -40,6 +41,7 @@
 import us.freeandfair.corla.model.Contest;
 import us.freeandfair.corla.model.Elector;
 import us.freeandfair.corla.model.UploadedFile;
+import us.freeandfair.corla.util.Pair;
 
 /**
  * Manages persistence through Hibernate, and provides several utility methods.
```

Comment:
The quick explanation: if you're doing one query that's built in, just do it and the right thing will happen; if you're doing multiple operations, do:

`boolean transaction = beginTransaction() 
... operations ... 
if (transaction) { logic to commitTransaction() or rollbackTransaction() as appropriate }```

(with appropriate exception handling, of course)

This is essentially to work around the fact that Hibernate doesn't support nested transactions, but all the queries and object updates need to happen in transactions. So if you want to do something, but don't know if there's already a transaction happening, `beginTransaction()` tells you (and starts one if there wasn't one).